### ISSUE TITLE
Strange Line Breaks

### ISSUE THREAD

**Issue by kerns** [khalwat/minify#3]
I am wrapping the entire contents of a layout template in minify tags – but the output is still over 500 lines long. Possibly attributable to breaks like these which seem to be added in following the opening of new tags.

![screenshot 2015-12-30 11 57 46](https://cloud.githubusercontent.com/assets/20254/12047916/d1e53c5c-aeec-11e5-857d-5936c8b0195b.png)

**Comment by khalwat** [comment]
This is actually an intentional effect of the Minify library that my Minify plugin is using.  If you copy that page source text out, and look at it carefully, you'll see that what's happened is where there used to be a space character after an opening tag (such as `<div`, it's been replaced with a line feed character `\n`

So where there used to be a space character, that space character has been removed, and replaced with a `\n`.

I can guess that they did this because it makes the text _somewhat_ more readable, while still being as compressed as possible.  Remember, the it doesn't matter how many _lines_ long the text is, it matters how many characters it's been minified down to.

It's valid as part of the HTML spec as well.  Kinda nifty.

**Comment by kerns** [comment]
OK – but I don't minify to make source code _more_ readable. On the contrary, I minify to obscure hacky and/or hastily written code. Oh, and performance.

**Comment by khalwat** [comment]
Here's the official answer from the Minify FAQ:

**Why do the CSS & HTML minifiers add so many line breaks?**

TL;DR: Ignore them. They don't add to the output size and if you absolutely want all content on one line you will have to use another tool.

It's rumored that some source control tools and old browsers don't like very long lines. Compressed files with shorter lines are also easier to diff.

Since both Minify classes are regex-based, it would be very difficult/error-prone to count characters then try to re-establish context to add line breaks. Instead, both classes trade 1 space for 1 line break (\n) wherever possible, adding breaks but without adding bytes.

If you can think of another safe & efficient way to limit lines in these two tools without adding bytes, please submit a patch, but this is not something anyone should be worrying about.

.....

I added this to Minify's README.md

**Comment by kerns** [comment]
I love that they felt compelled to dedicate four paragraphs to this – must be a lot of other developers with low-level OCD frustrated not to see one long single line of code.

> Compressed files with shorter lines are also easier to diff.

But I don't buy this – or at least, it doesn't seem to be a issue for extremely large chunks of JS. Also, I guess there could be a good use case for needing to diff compressed code, but it might also be an edge case.

Anyway. Thanks! I'll live with it. ;-)

**Comment by khalwat** [comment]
yeah it sounds like he was annoyed by the number of people saying "but how come my minified HTML isn't minified, it has line breaks!" or such.

It's a natural question, I wondered what was going on as well when I first saw it.  But it's perfectly harmless.  The main objective of minimizing the code is met.
